src/transformers/modeling_bart.py

```diff
@@ -981,7 +981,7 @@ def prepare_inputs_for_generation(self, decoder_input_ids, past, attention_mask,
         }
 
     def prepare_scores_for_generation(self, scores, cur_len, max_length):
-        if cur_len == 1:
```

Comment:
This would break the `Bart` tests - why should we change that here?

Reply:
Sorry, forgot about this PR, should be good now.

Regarding line 984 in modeling_bart.py, it might be wrong because I could not get beam search working with a vanilla Bart model. Where non-beam .generate creates sequences starting with a single BOS token, that line in Bart implementation creates sequences with two BOS tokens at the start. Changing that line fixed it for me but then maybe I'm not doing something right. I'll see if I should create an issuse.